webprotege-client/src/main/java/edu/stanford/bmir/protege/web/client/merge_add/UploadAndMergeAdditionsProjectsWorkflow.java

```diff
@@ -0,0 +1,108 @@
+
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.List;
+
+public class UploadAndMergeAdditionsProjectsWorkflow {
+
+    @Nonnull
+    private final SelectOptionForMergeAdditionsWorkflow selectOptionsWorkflow;
+
+    @Nonnull
+    private final DispatchServiceManager dispatchServiceManager;
+
+    @Nonnull
+    private final DispatchErrorMessageDisplay errorDisplay;
+
+    @Nonnull
+    private final ProgressDisplay progressDisplay;
+
+    @Inject
+    public UploadAndMergeAdditionsProjectsWorkflow(@Nonnull SelectOptionForMergeAdditionsWorkflow selectOptionsWorkflow,
+                                                   @Nonnull DispatchServiceManager dispatchServiceManager,
+                                                   @Nonnull DispatchErrorMessageDisplay errorDisplay,
+                                                   @Nonnull ProgressDisplay progressDisplay) {
+        this.selectOptionsWorkflow = selectOptionsWorkflow;
+        this.dispatchServiceManager = dispatchServiceManager;
+        this.errorDisplay = errorDisplay;
+        this.progressDisplay = progressDisplay;
+    }
+
+    public void start(ProjectId projectId) {
+        uploadProject(projectId);
+    }
+
+    private void uploadProject(final ProjectId projectId) {
+        UploadFileDialogController uploadFileDialogController = new UploadFileDialogController("Upload ontologies", new UploadFileResultHandler() {
+            @Override
+            public void handleFileUploaded(DocumentId fileDocumentId) {
+                getOntologies(projectId, fileDocumentId);
+            }
+
+            @Override
+            public void handleFileUploadFailed(String errorMessage) {
+                GWT.log("Upload failed");
+            }
+        });
+        WebProtegeDialog.showDialog(uploadFileDialogController);
+    }
+
+    private void getOntologies(ProjectId projectId, DocumentId documentId){
+        dispatchServiceManager.execute(new GetAllOntologiesAction(projectId, documentId), new DispatchServiceCallbackWithProgressDisplay<GetAllOntologiesResult>(errorDisplay, progressDisplay) {
+            @Override
+            public String getProgressDisplayTitle() {
+                return "Uploading Ontologies";
+            }
+
+            @Override
+            public String getProgressDisplayMessage() {
+                return "Uploading  and Getting Ontologies. Please Wait.";
+            }
+
+            @Override
+            public void handleSuccess(GetAllOntologiesResult result){
+                selectOntologies(projectId, documentId, result);
+            }
+        });
+    }
+
+    private void selectOntologies(ProjectId projectId, DocumentId documentId, GetAllOntologiesResult result){
+        ArrayList<OWLOntologyID> list = (ArrayList<OWLOntologyID>) result.getOntologies();
+
+        SelectOntologiesForMergeView view = new SelectOntologiesForMergeViewImpl(list);
```

Comment:
This is optional. 

Rather than the dialog controller, take a look at the ModalPresenter and use this (see examples throughout the code base).  If you don't want to take this on, I understand and we'll still accept your pull request as it is.